source/common/event/scaled_range_timer_manager_impl.cc

```diff
@@ -123,60 +123,105 @@ class ScaledRangeTimerManager::RangeTimerImpl final : public RangeTimer {
     }
   }
 
-  ScaledRangeTimerManager& manager_;
+  ScaledRangeTimerManagerImpl& manager_;
   const TimerCb callback_;
   const TimerPtr min_duration_timer_;
 
   absl::variant<Inactive, WaitingForMin, ScalingMax> state_;
   const ScopeTrackedObject* scope_;
 };
 
-ScaledRangeTimerManager::ScaledRangeTimerManager(Dispatcher& dispatcher)
+class ScaledRangeTimerManagerImpl::FixedMinimumRangeTimerImpl final : public Event::Timer {
+public:
+  FixedMinimumRangeTimerImpl(ScaledTimerMinimum minimum, TimerCb timer_cb,
+                             ScaledRangeTimerManagerImpl& manager)
+      : minimum_(minimum), range_timer_(std::move(timer_cb), manager) {}
+
+  void enableTimer(std::chrono::milliseconds ms,
+                   const ScopeTrackedObject* object = nullptr) override {
+    range_timer_.enableTimer(computeMinimum(minimum_, ms), ms, object);
+  }
+
+  void enableHRTimer(std::chrono::microseconds us,
+                     const ScopeTrackedObject* object = nullptr) override {
+    enableTimer(std::chrono::duration_cast<std::chrono::milliseconds>(us), object);
```

Comment:
I think you should implement enableTimer in terms of enableHRTimer to avoid loss of precision.  That said, I should try to remove the enableHRTimer from the timer interface; it should be possible now that enableTimer(zeroDuration) doesn't result in a timer that fires in the same event loop iteration.